extensions/amp-form/0.1/amp-form.js

```diff
@@ -807,6 +822,15 @@ function updateInvalidTypesClasses(element) {
   }
 }
 
+function removeValidityStateClasses(form) {
+  const dummyInput = document.createElement('input');
+  for (const validityState in dummyInput.validity) {
```

Comment:
We can pre-compute this outside the method to avoid repeated DOM element allocations.

Reply:
I am unable to understand, can you please explain briefly ?